File: Phython GITS assignment /Python 2026/Logistics.py

base_pay = 5000
 
def collection_table():
    print("""              
                    BACK TO SENDER LOGISTICS SERVICES COMMISSION TABLE
               ------------------------------------------------------------              
              | Collection Rate |    Amount per Parcel     |   Base Pay   |
              -------------------------------------------------------------                      
  
              |Less than 50%    |         ₦160             |    ₦5,000 
             
              |50% - 59%        |         ₦200             |    ₦5,000  

              |60% - 69%        |         ₦250             |    ₦5,000  
             
              |>=70%            |         ₦500             |    ₦5,000 
              -------------------------------------------------------------
        """)

collection_table()



riders_name = input("\nEnter Riders Name: ").strip()
print(f"\nDear {riders_name}, Lets check the tasks you made today")

#if riders_name = " ":
#    print("invalid name, enter a name")
#    exit()

collection_rate = int(input("\nEnter Number Of Succcessfull Delivery Done Today: "))

if collection_rate < 0 or collection_rate > 100:
    print("\nInvalid Entry! Collection rate must be between 0 and 100.")
    exit()


def collection_pay(collection_rate):

    if collection_rate >= 70:
        
        amount_per_parcel = 500
             
    elif collection_rate >= 60 and collection_rate <= 69:

        amount_per_parcel = 250

    elif collection_rate >= 50 and collection_rate <= 59:

        amount_per_parcel = 200
    
    else:
        amount_per_parcel = 160

    expected_returns = base_pay + (collection_rate * amount_per_parcel) 
    return expected_returns

expected_returns = collection_pay(collection_rate)

print(f"""\n Dear {riders_name},
 Successful Deliveries: {collection_rate} 
 Total Earnings: ₦{expected_returns} 
""")

File: JAVA GITS ASSINGMENT/2026 project/Logistics.java
import java.util.Scanner;
public class Logistics{
public static void main (String[] args){
Scanner input = new Scanner (System.in);

int basePay = 5000;


boolean running = true;


while (running){

for(int index = 0; index <=48; index++){
System.out.print("=");
}
System.out.println("\n Collection Rate   Amount Per Parcel   Base Pay");

for(int index = 0; index <=48; index++){    
System.out.print("=");
}


System.out.println("\n Less than 50%\t\t160\t\t5000\n 50-59%\t\t\t200\t\t5000\n 60-69%\t\t\t250\t\t5000\n >=70%\t\t\t500\t\t5000");

System.out.println("\nEnter Rider Name: ");
String riderName = input.nextLine();

System.out.println("\nEnter number of delivery: ");
int delivery = input.nextInt();

int expectedResult2 = 0;



if(delivery < 0 || delivery > 100){
System.out.print("Invalid input");


}else if(delivery < 50){
      int amount_per_parcel = 160;
        expectedResult2 = basePay + (delivery * amount_per_parcel);

 
}else if(delivery >= 50 && delivery <= 59){
      int amount_per_parcel = 200;
        expectedResult2 = basePay + (delivery * amount_per_parcel);


}else if(delivery >= 60 && delivery <= 69){
      int amount_per_parcel = 250;
        expectedResult2 = basePay + (delivery * amount_per_parcel);


}else if(delivery >= 70){
      int amount_per_parcel = 500;
        expectedResult2 = basePay + (delivery * amount_per_parcel);


}
input.nextLine();


System.out.printf("\nDear %s,Congratulations! You've Successfully made %d delivery today and your expected wage is %d ", riderName, delivery, expectedResult2 );


System.out.println("\nWill you like to continue, enter yes/no? ");

String enter = input.nextLine().toLowerCase().trim();



if(enter.equals ("no")){
break;

}else if(enter.equals ("yes")){
continue;
}else if(!enter.equals ("no") || !enter.equals ("yes")){
    System.out.println("Invalid entry");
   continue;
}

}
}
}

